lib/redis.js

```diff
@@ -95,11 +95,12 @@ module.exports = CoreObject.extend({
 
   _updateRecentUploadsList: function(keyPrefix, revisionKey) {
     var client = this._client;
-    return client.lpush(keyPrefix, revisionKey);
+    var score = new Date().getTime();
+    return client.zadd(keyPrefix, score, revisionKey);
   },
 
   _trimRecentUploadsList: function(keyPrefix, maxEntries) {
     var client = this._client;
-    return client.ltrim(keyPrefix, 0, maxEntries - 1);
+    return client.zremrangebyrank(keyPrefix, 0, -(maxEntries + 1));
```

Comment:
yes, but in this PR we're still not trimming the keys that have the index so it won't make a difference for now,

my plan was to get this merged first and then do what we discussed on slack to do the houskeeping of index keys,

for now this has the same behaviour as the old code, if you like we can do also the next steps in this PR instead